apiserver/facades/agent/storageprovisioner/storageprovisioner_caas_test.go

```diff
@@ -42,41 +47,31 @@ func TestCaasProvisionerSuite(t *testing.T) {
 func (s *caasProvisionerSuite) TestWatchApplications(c *tc.C) {
 	defer s.setupMocks(c).Finish()
 
+	done := make(chan struct{})
+	defer close(done)
 	ch := make(chan []string)
 
-	watcher := watchertest.NewMockStringsWatcher(ch)
-	s.backend.EXPECT().WatchApplications().DoAndReturn(func() state.StringsWatcher {
-		// Enqueue
-		go func() {
-			select {
-			case ch <- []string{"application-mariadb"}:
-			case <-time.After(testhelpers.LongWait):
-				c.Fatalf("timed out waiting to send")
-			}
-		}()
-		return watcher
-	})
-	s.resources.EXPECT().Register(watcher).Return("1")
+	w := watchertest.NewMockStringsWatcher(ch)
+	s.applicationService.EXPECT().WatchApplications(gomock.Any()).
+		DoAndReturn(func(context.Context) (watcher.Watcher[[]string], error) {
+			time.AfterFunc(internaltesting.ShortWait, func() {
+				// Send initial event.
+				select {
+				case ch <- []string{"application-mariadb"}:
+				case <-done:
+					c.Error("watcher (applications) did not fire")
+				}
+			})
+			return w, nil
+		})
+	s.watcherRegistry.EXPECT().Register(gomock.Any()).Return("1", nil)
```

Comment:
I think we have some tooling, so you don't need to do this kind of thing?

@SimonRichardson do you know?

Reply:
Were you referring to `EnsureRegisterWatcher`? The tests don't change though.